Description: Check for file existence in packageFiles instead of matching strs
Author: Doug Torrance <dtorrance@debian.org>
Origin: https://github.com/Macaulay2/M2/pull/3864
Last-Update: 2025-06-02

--- a/M2/Macaulay2/m2/packages.m2
+++ b/M2/Macaulay2/m2/packages.m2
@@ -556,11 +556,7 @@
 
 packageFiles = pkg -> (
     srcfile := realpath pkg#"source file";
-    if (
-	srcfile == "stdio"         or
-	srcfile == "currentString" or
-	match("/startup\\.m2(?:\\.in)?$", srcfile))
-    then {}
+    if not fileExists srcfile then {}
     else prepend(srcfile,
 	if not pkg#?"auxiliary files" then {}
 	else select(values loadedFiles, match_(pkg#"auxiliary files"))))
